core/src/test/java/com/linecorp/armeria/internal/common/stream/FixedStreamMessageTest.java

```diff
@@ -74,15 +78,65 @@ public void onComplete() {}
         assertThat(received).hasValue(0);
     }
 
-    private static final class IntsProvider implements ArgumentsProvider {
+    @ArgumentsSource(FixedStreamMessageProvider.class)
+    @ParameterizedTest
+    void raceBetweenSubscriptionAndAbort(StreamMessage<Integer> stream) {
+        final CountDownLatch latch = new CountDownLatch(1);
+        final AtomicReference<Throwable> causeRef = new AtomicReference<>();
+        stream.subscribe(new Subscriber<Integer>() {
```

Comment:
Do you mean using `collect`?
I'm asking because the commit description the situation when `abort` and `collect` is called but I don't see the test that uses `collect`. Also, this test seems like passing without the change. 🤔

Reply:
I tried to make a race condition between `collect()` and `abort()` but I couldn't come up with a good idea. I added a test case with `Subscriber` API to make sure change to work.
`StreamMessageCollectingTest` checks the basic operations.

Reply:
Yeah, it's really hard to make a test for such a case. 👍